modules/sfp_neutrinoapi.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_neutrinoapi
+# Purpose:      SpiderFoot plug-in to search NeutrinoAPI for IP address info and
+#               check IP reputation.
+#
+# Author:      Brendan Coles <bcoles@gmail.com>
+#
+# Created:     2018-11-30
+# Copyright:   (c) Brendan Coles 2018
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import json
+import socket
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+
+class sfp_neutrinoapi(SpiderFootPlugin):
+    """NeutrinoAPI:Footprint,Investigate,Passive:Reputation Systems:apikey:Search NeutrinoAPI for IP address info and check IP reputation."""
+
+    # Default options
+    opts = {
+        'user_id': '',
+        'api_key': '',
+        'timeout': 30
+    }
+
+    # Option descriptions
+    optdescs = {
+        'user_id': "NeutrinoAPI user ID.",
+        'api_key': "NeutrinoAPI API key.",
+        'timeout': "Query timeout, in seconds."
+    }
+
+    results = dict()
+    errorState = False
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.__dataSource__ = "NeutrinoAPI"
```

Comment:
Just FYI for the future, you only need to set this if you intend to override the module name, otherwise the module name is used by default as the data source.